paymentsheet/src/main/java/com/stripe/android/lpmfoundations/luxe/FormElementsBuilder.kt

```diff
@@ -0,0 +1,91 @@
+package com.stripe.android.lpmfoundations.luxe
+
+import com.stripe.android.core.model.CountryUtils
+import com.stripe.android.lpmfoundations.paymentmethod.UiDefinitionFactory
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.ui.core.elements.AddressSpec
+import com.stripe.android.uicore.elements.FormElement
+
+internal class FormElementsBuilder(
+    private val arguments: UiDefinitionFactory.Arguments,
+) {
+    private val headerFormElements: MutableList<FormElement> = mutableListOf()
+    private val uiFormElements: MutableList<FormElement> = mutableListOf()
+    private val footerFormElements: MutableList<FormElement> = mutableListOf()
+
+    private val requiredContactInformationCollectionModes: MutableSet<ContactInformationCollectionMode> = mutableSetOf()
+
+    private var requireBillingAddressCollection: Boolean = false
+    private var availableCountries: Set<String> = CountryUtils.supportedBillingCountries
+
+    init {
+        // Setup the required contact information fields based on the merchant billingDetailsCollectionConfiguration.
+        for (value in ContactInformationCollectionMode.entries) {
+            if (value.isRequired(arguments.billingDetailsCollectionConfiguration)) {
+                requireContactInformation(value)
+            }
+        }
+
+        // Setup the required billing fields section based on the merchant billingDetailsCollectionConfiguration.
+        if (arguments.billingDetailsCollectionConfiguration.address
+            == PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode.Full
+        ) {
+            requireBillingAddress()
+        }
+    }
+
+    fun header(formElement: FormElement): FormElementsBuilder = apply {
+        headerFormElements += formElement
+    }
+
+    fun requireContactInformation(type: ContactInformationCollectionMode): FormElementsBuilder = apply {
+        if (type.isAllowed(arguments.billingDetailsCollectionConfiguration)) {
+            requiredContactInformationCollectionModes += type
+        }
+    }
+
+    fun element(formElement: FormElement): FormElementsBuilder = apply {
+        uiFormElements += formElement
+    }
+
+    fun requireBillingAddress(
```

Comment:
what do you think of calling this something like `maybeRequireBillingAddress`? I think it's a little confusing that this is called `requireBillingAddress` but in some cases, you could call this function and the billing address would still not be added (if I understand correctly)

Reply:
same thought for `requireContactInformation`

Reply:
How about `requireBillingAddressIfAllowed`?

Reply:
sgtm!